repo2docker/buildpacks/repo2docker-entrypoint

```diff
@@ -23,11 +23,13 @@ SIGNALS = set(signal.Signals) - {signal.SIGKILL, signal.SIGSTOP, signal.SIGCHLD}
 
 def main():
 
-    # open log file to send output
-    log_file = open(
-        os.path.join(os.environ.get("REPO_DIR", "."), ".jupyter-server-log.txt"),
-        "ab",
-    )
+    # open log file to send output; fall back to writing it in
+    # the current working directory
+    if "REPO_DIR" in os.environ and os.access(os.environ["REPO_DIR"], 7):
```

Comment:
It's still possible to fail on `.`. Maybe iterate through `$REPO_DIR, ., /tmp`, breaking when access(os.W_OK) passes?

Perhaps simplest is to attempt the open itself until it succeeds:

```python
for d in [$repo_dir, ., tmp]:
    log_path = join(d, 'log.txt')
    try:
        log_file = open(log_path, "ab")
    except PermissionError:
        continue
    else:
        # success
        break
```